src/CamOverlayDrawingAPI.ts

```diff
@@ -0,0 +1,252 @@
+                port: this.port,
+                address: '/local/camoverlay/ws',
+                protocol: 'cairo-api',
+
+                auth: this.auth,
+                tls: this.tls,
+                tlsInsecure: this.tlsInsecure,
+            };
+            this.ws = new WsClient(options);
+
+            this.ws.on('open', () => {
+                this.reportMessage('Websocket opened');
+                resolve();
+            });
+            this.ws.on('message', (data: string) => {
+                let dataJSON = JSON.parse(data);
+                if (dataJSON.hasOwnProperty('call_id') && dataJSON['call_id'] in this.sendMessages) {
+                    if (dataJSON.hasOwnProperty('error')) {
+                        this.sendMessages[dataJSON['call_id']].reject(new Error(dataJSON.error));
+                    } else {
+                        this.sendMessages[dataJSON['call_id']].resolve(dataJSON);
+                    }
+                    delete this.sendMessages[dataJSON['call_id']];
+                }
+
+                if (dataJSON.hasOwnProperty('error')) {
+                    this.reportError(new Error(dataJSON.error));
+                } else {
+                    this.reportMessage(data);
+                }
+            });
+            this.ws.on('error', (error: Error) => {
+                this.reportError(error);
+                reject(error);
+            });
+            this.ws.on('close', () => {
+                this.reportClose();
+            });
+
+            this.ws.open();
+        });
+    }
+
+    private reportMessage(msg: string) {
+        this.emit('message', msg);
+    }
+
+    private reportError(err: Error) {
+        this.ws.close();
+        this.emit('error', err);
+    }
+
+    private reportClose() {
+        for (const callId in this.sendMessages) {
+            this.sendMessages[callId].reject(new Error('Connection lost'));
+        }
+        this.sendMessages = {};
+        this.emit('close');
+    }
+
+    cairo(command: string, ...params: any[]) {
+        return this.sendMessage({ command: command, params: params }) as Promise<CairoResponse | CairoCreateResponse>;
+    }
+
+    writeText(...params: WriteTextParams) {
+        return this.sendMessage({ command: 'write_text', params: params }) as Promise<CairoResponse>;
+    }
+
+    uploadImageData(imgBuffer: Buffer) {
+        return this.sendBinaryMessage(
+            {
+                command: 'upload_image_data',
+                params: [],
+            },
+            imgBuffer
+        ) as Promise<UploadImageResponse>;
+    }
+
+    uploadFontData(fontBuffer: Buffer) {
+        return this.sendBinaryMessage(
+            {
+                command: 'upload_font_data',
+                params: [fontBuffer.toString('base64')],
+            },
+            fontBuffer
+        ) as Promise<CairoCreateResponse>;
+    }
+
+    showCairoImage_v2(cairoImage: string, posX: number, posY: number, zIndex: number) {
```

Comment:
tyhle nazvy funkce bych asi nemenil..to prave odstini skripty od interni zmeny

Reply:
!